File: front/central-next-js/src/features/business/domain/Business.ts
// Domain - Business Entity
export interface Business {
  id: number;
  name: string;
  code: string;
  description: string;
  address: string;
  phone: string;
  email: string;
  website: string;
  logo_url: string;
  timezone: string;
  primary_color: string;
  secondary_color: string;
  custom_domain: string;
  is_active: boolean;
  business_type_id: number;
  business_type: string;
  enable_delivery: boolean;
  enable_pickup: boolean;
  enable_reservations: boolean;
  created_at: string;
  updated_at: string;
}

export interface BusinessType {
  id: number;
  name: string;
  code: string;
  description: string;
  icon: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface CreateBusinessRequest {
  name: string;
  code: string;
  description: string;
  address: string;
  phone?: string;
  email?: string;
  website?: string;
  logo_url?: string;
  logo_file?: File;
  timezone?: string;
  primary_color?: string;
  secondary_color?: string;
  custom_domain?: string;
  business_type_id: number;
  enable_delivery?: boolean;
  enable_pickup?: boolean;
  enable_reservations?: boolean;
}

export interface UpdateBusinessRequest {
  name?: string;
  code?: string;
  description?: string;
  address?: string;
  phone?: string;
  email?: string;
  website?: string;
  logo_url?: string;
  logo_file?: File;
  timezone?: string;
  primary_color?: string;
  secondary_color?: string;
  custom_domain?: string;
  business_type_id?: number;
  is_active?: boolean;
  enable_delivery?: boolean;
  enable_pickup?: boolean;
  enable_reservations?: boolean;
}

export interface CreateBusinessTypeRequest {
  name: string;
  code: string;
  description: string;
  icon: string;
}

export interface UpdateBusinessTypeRequest {
  name?: string;
  code?: string;
  description?: string;
  icon?: string;
  isActive?: boolean;
}

// API Response types
export interface BusinessResponse {
  success: boolean;
  message: string;
  data: Business | Business[];
}

export interface BusinessTypeResponse {
  success: boolean;
  message: string;
  data: BusinessType | BusinessType[];
}

export interface BusinessListDTO {
  businesses: Business[];
  total: number;
  page: number;
  limit: number;
}

File: back/central-reserve/internal/domain/dtos/business.go
package dtos

import (
	"mime/multipart"
	"time"
)

// BusinessTypeRequest representa la solicitud para crear/actualizar un tipo de negocio
type BusinessTypeRequest struct {
	Name        string
	Code        string
	Description string
	Icon        string
	IsActive    bool
}

// BusinessTypeResponse representa la respuesta de un tipo de negocio
type BusinessTypeResponse struct {
	ID          uint
	Name        string
	Code        string
	Description string
	Icon        string
	IsActive    bool
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

// BusinessRequest representa la solicitud para crear/actualizar un negocio
type BusinessRequest struct {
	Name           string
	Code           string
	BusinessTypeID uint
	Timezone       string
	Address        string
	Description    string

	// Configuración de marca blanca
	LogoFile        *multipart.FileHeader // Archivo de imagen para subir a S3
	PrimaryColor    string
	SecondaryColor  string
	TertiaryColor   string
	QuaternaryColor string
	NavbarImageFile *multipart.FileHeader // Imagen de navbar para subir a S3
	CustomDomain    string
	IsActive        bool

	// Configuración de funcionalidades
	EnableDelivery     bool
	EnablePickup       bool
	EnableReservations bool
}

// UpdateBusinessRequest representa actualización parcial de negocio
// Todos los campos son opcionales; los nil no modifican
type UpdateBusinessRequest struct {
	Name               *string
	Code               *string
	BusinessTypeID     *uint
	Timezone           *string
	Address            *string
	Description        *string
	LogoFile           *multipart.FileHeader
	PrimaryColor       *string
	SecondaryColor     *string
	TertiaryColor      *string
	QuaternaryColor    *string
	NavbarImageFile    *multipart.FileHeader
	CustomDomain       *string
	IsActive           *bool
	EnableDelivery     *bool
	EnablePickup       *bool
	EnableReservations *bool
}

// BusinessResponse representa la respuesta de un negocio
type BusinessResponse struct {
	ID           uint
	Name         string
	Code         string
	BusinessType BusinessTypeResponse
	Timezone     string
	Address      string
	Description  string

	// Configuración de marca blanca
	LogoURL         string
	PrimaryColor    string
	SecondaryColor  string
	TertiaryColor   string
	QuaternaryColor string
	NavbarImageURL  string
	CustomDomain    string
	IsActive        bool

	// Configuración de funcionalidades
	EnableDelivery     bool
	EnablePickup       bool
	EnableReservations bool

	CreatedAt time.Time
	UpdatedAt time.Time
}

// BusinessListResponse representa la respuesta de lista de negocios
type BusinessListResponse struct {
	Businesses []BusinessResponse
	Total      int64
	Page       int
	Limit      int
}

// BusinessTypeListResponse representa la respuesta de lista de tipos de negocio
type BusinessTypeListResponse struct {
	BusinessTypes []BusinessTypeResponse
	Total         int64
	Page          int
	Limit         int
}

// BusinessResourceConfiguredResponse representa la respuesta de recursos configurados de un negocio
type BusinessResourceConfiguredResponse struct {
	ResourceID   uint
	ResourceName string
	IsActive     bool
}

// BusinessResourcesResponse representa la respuesta completa de recursos de un negocio
type BusinessResourcesResponse struct {
	BusinessID uint
	Resources  []BusinessResourceConfiguredResponse
	Total      int
	Active     int
	Inactive   int
}
